### PR TITLE
SUBMARINE-1329. Improve test coverage of submarine.artifacts module

### PR DESCRIPTION
**Pull request by huang06**
### What is this PR for?

We lack tests in the `submodule.artifacts` module. We should add tests and aim for at least 90% test coverage for this module.

**Additionally**, this PR also fix SUBMARINE-922 issue since #994 accidentally modify the code.

### What type of PR is it?

Improvement

### What is the Jira issue?

<https://issues.apache.org/jira/browse/SUBMARINE-1329>

### How should this be tested?

```bash
cd submarine-sdk/pysubmarine
python3 -m venv .venv
source .venv/bin/activate
pip install -U pip setuptools wheel
pip install -e .[tf2,pytorch]
pip install -r github-actions/test-requirements.txt
```

```bash
$ pytest -m 'not e2e'

...
---------- coverage: platform linux, python 3.7.14-final-0 -----------
Name                                                        Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------------------
submarine/__init__.py                                           9      0   100%
submarine/artifacts/__init__.py                                 2      0   100%
submarine/artifacts/repository.py                              30      0   100%
submarine/cli/__init__.py                                       2      0   100%
...
-----------------------------------------------------------------------------------------
TOTAL                                                        4725   2621    45%

============================================================================ 42 passed, 7 skipped, 31 deselected, 10 warnings in 23.88s ============================================================================
```

### Questions:
* Do the license files need updating? Yes/No
* Are there breaking changes for older versions? Yes/No
* Does this need new documentation? Yes/No

### CODE REVIEW

File: submarine-sdk/pysubmarine/pytest.ini [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
[the remark this answers is not in the document]
Comment (huang06) [reply submarine-sdk/pysubmarine/pytest.ini:1]: By default, the additional files, such as `pytest.ini`, `pylintrc`, would not been installed into the wheel file.

```bash
cd submarine-sdk/pysubmarine
python3 -m venv .venv
source .venv/bin/activate
python3 -m pip install -U pip setuptools wheel build
```

```bash
$ python3 -m build
...
Successfully built apache-submarine-0.8.0-SNAPSHOT.tar.gz and apache_submarine-0.8.0_SNAPSHOT-py3-none-any.whl
```

```bash
$ python3 -c '
import pprint
from zipfile import ZipFile
                 
path = "dist/apache_submarine-0.8.0_SNAPSHOT-py3-none-any.whl"
names = ZipFile(path).namelist()
pprint.pprint(names)
'

['submarine/__init__.py',
 'submarine/exceptions.py',
 'submarine/artifacts/__init__.py',
 'submarine/artifacts/repository.py',
 'submarine/cli/__init__.py',
 'submarine/cli/main.py',
 'submarine/cli/config/__init__.py',
 'submarine/cli/config/cli_config.yaml',
 'submarine/cli/config/command.py',
 'submarine/cli/config/config.py',
 'submarine/cli/environment/__init__.py',
 'submarine/cli/environment/command.py',
 'submarine/cli/experiment/__init__.py',
 'submarine/cli/experiment/command.py',
 'submarine/cli/notebook/__init__.py',
 'submarine/cli/notebook/command.py',
 'submarine/cli/sandbox/__init__.py',
 'submarine/cli/sandbox/command.py',
 'submarine/cli/serve/__init__.py',
 'submarine/cli/serve/command.py',
 'submarine/client/__init__.py',
 'submarine/client/api_client.py',
 'submarine/client/configuration.py',
 'submarine/client/exceptions.py',
 'submarine/client/rest.py',
 'submarine/client/api/__init__.py',
 'submarine/client/api/environment_api.py',
 'submarine/client/api/environment_client.py',
 'submarine/client/api/experiment_api.py',
 'submarine/client/api/experiment_client.py',
 'submarine/client/api/notebook_api.py',
 'submarine/client/api/notebook_client.py',
 'submarine/client/api/serve_api.py',
 'submarine/client/api/serve_client.py',
 'submarine/client/models/__init__.py',
 'submarine/client/models/code_spec.py',
 'submarine/client/models/environment_spec.py',
 'submarine/client/models/experiment_meta.py',
 'submarine/client/models/experiment_spec.py',
 'submarine/client/models/experiment_task_spec.py',
 'submarine/client/models/experiment_template_submit.py',
 'submarine/client/models/json_response.py',
 'submarine/client/models/kernel_spec.py',
 'submarine/client/models/notebook_meta.py',
 'submarine/client/models/notebook_pod_spec.py',
 'submarine/client/models/notebook_spec.py',
 'submarine/client/models/serve_request.py',
 'submarine/client/models/serve_spec.py',
 'submarine/client/utils/__init__.py',
 'submarine/client/utils/api_utils.py',
 'submarine/entities/Metric.py',
 'submarine/entities/Param.py',
 'submarine/entities/__init__.py',
 'submarine/entities/_submarine_object.py',
 'submarine/entities/experiment.py',
 'submarine/entities/model_registry/__init__.py',
 'submarine/entities/model_registry/model_stages.py',
 'submarine/entities/model_registry/model_version.py',
 'submarine/entities/model_registry/model_version_tag.py',
 'submarine/entities/model_registry/registered_model.py',
 'submarine/entities/model_registry/registered_model_tag.py',
 'submarine/ml/__init__.py',
 'submarine/ml/abstract_model.py',
 'submarine/ml/pytorch/__init__.py',
 'submarine/ml/pytorch/loss.py',
 'submarine/ml/pytorch/metric.py',
 'submarine/ml/pytorch/optimizer.py',
 'submarine/ml/pytorch/parameters.py',
 'submarine/ml/pytorch/registries.py',
 'submarine/ml/pytorch/input/__init__.py',
 'submarine/ml/pytorch/input/libsvm_dataset.py',
 'submarine/ml/pytorch/layers/__init__.py',
 'submarine/ml/pytorch/layers/core.py',
 'submarine/ml/pytorch/model/__init__.py',
 'submarine/ml/pytorch/model/base_pytorch_model.py',
 'submarine/ml/pytorch/model/ctr/__init__.py',
 'submarine/ml/pytorch/model/ctr/afm.py',
 'submarine/ml/pytorch/model/ctr/deepfm.py',
 'submarine/ml/tensorflow/__init__.py',
 'submarine/ml/tensorflow/optimizer.py',
 'submarine/ml/tensorflow/parameters.py',
 'submarine/ml/tensorflow/registries.py',
 'submarine/ml/tensorflow/input/__init__.py',
 'submarine/ml/tensorflow/input/input.py',
 'submarine/ml/tensorflow/layers/__init__.py',
 'submarine/ml/tensorflow/layers/core.py',
 'submarine/ml/tensorflow/model/__init__.py',
 'submarine/ml/tensorflow/model/base_tf_model.py',
 'submarine/ml/tensorflow/model/ccpm.py',
 'submarine/ml/tensorflow/model/deepfm.py',
 'submarine/ml/tensorflow/model/fm.py',
 'submarine/ml/tensorflow/model/nfm.py',
 'submarine/ml/tensorflow_v2/__init__.py',
 'submarine/ml/tensorflow_v2/optimizer.py',
 'submarine/ml/tensorflow_v2/parameters.py',
 'submarine/ml/tensorflow_v2/registries.py',
 'submarine/ml/tensorflow_v2/input/__init__.py',
 'submarine/ml/tensorflow_v2/input/input.py',
 'submarine/ml/tensorflow_v2/layers/__init__.py',
 'submarine/ml/tensorflow_v2/layers/core.py',
 'submarine/ml/tensorflow_v2/model/__init__.py',
 'submarine/ml/tensorflow_v2/model/base_tf_model.py',
 'submarine/ml/tensorflow_v2/model/ccpm.py',
 'submarine/ml/tensorflow_v2/model/deepfm.py',
 'submarine/ml/tensorflow_v2/model/fm.py',
 'submarine/ml/tensorflow_v2/model/nfm.py',
 'submarine/models/__init__.py',
 'submarine/models/pytorch.py',
 'submarine/models/tensorflow.py',
 'submarine/store/__init__.py',
 'submarine/store/database/__init__.py',
 'submarine/store/database/db_types.py',
 'submarine/store/database/models.py',
 'submarine/store/model_registry/__init__.py',
 'submarine/store/model_registry/abstract_store.py',
 'submarine/store/model_registry/sqlalchemy_store.py',
 'submarine/store/tracking/__init__.py',
 'submarine/store/tracking/abstract_store.py',
 'submarine/store/tracking/sqlalchemy_store.py',
 'submarine/tracking/__init__.py',
 'submarine/tracking/client.py',
 'submarine/tracking/constant.py',
 'submarine/tracking/fluent.py',
 'submarine/tracking/utils.py',
 'submarine/utils/__init__.py',
 'submarine/utils/db_utils.py',
 'submarine/utils/env.py',
 'submarine/utils/fileio.py',
 'submarine/utils/pytorch_utils.py',
 'submarine/utils/rest_utils.py',
 'submarine/utils/tf_utils.py',
 'submarine/utils/tf_utils_v2.py',
 'submarine/utils/validation.py',
 'apache_submarine-0.8.0_SNAPSHOT.dist-info/METADATA',
 'apache_submarine-0.8.0_SNAPSHOT.dist-info/WHEEL',
 'apache_submarine-0.8.0_SNAPSHOT.dist-info/entry_points.txt',
 'apache_submarine-0.8.0_SNAPSHOT.dist-info/top_level.txt',
 'apache_submarine-0.8.0_SNAPSHOT.dist-info/RECORD']
```

[the remark this answers is not in the document]
Comment (cdmikechen) [reply submarine-sdk/pysubmarine/pytest.ini]: If it is confirmed, then LGTM

[the remark this answers is not in the document]
Comment (cdmikechen) [reply submarine-sdk/pysubmarine/pytest.ini]: I will create another issue to consider solving the problem of too much workflow causing too long check times.

**Review by cdmikechen — APPROVED** [review]
LGTM.
And I create a new issue https://issues.apache.org/jira/browse/SUBMARINE-1330 to discuss and refine the cicd workflow.

**Pull request closed without merging**